packages/smooth_app/lib/helpers/picture_capture_helper.dart

```diff
@@ -19,13 +23,62 @@ Future<File?> pickImageAndCrop() async {
     return null;
   }
 
-  return ImageCropper.cropImage(
-    sourcePath: pickedXFile.path,
-    androidUiSettings: const AndroidUiSettings(
-      lockAspectRatio: false,
-      hideBottomControls: true,
+  Uint8List? bytes = await pickedXFile.readAsBytes();
+  final CropController _controller = CropController();
+
+  await Navigator.push<Widget>(
+    context,
+    MaterialPageRoute<Widget>(
+      builder: (BuildContext context) {
+        context.watch<ThemeProvider>();
+        final ThemeData theme = Theme.of(context);
+        return Scaffold(
+          body: Crop(
+            image: bytes!,
+            controller: _controller,
+            onCropped: (Uint8List image) {
+              bytes = image;
+              Navigator.pop(context);
+            },
+            initialSize: 0.5,
+            baseColor: theme.colorScheme.primary,
+            maskColor: Colors.white.withAlpha(100),
+            cornerDotBuilder: (double size, EdgeAlignment edgeAlignment) =>
+                DotControl(color: theme.colorScheme.primary),
+          ),
+          bottomNavigationBar: BottomAppBar(
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: <Widget>[
+                IconButton(
+                  icon: const Icon(Icons.cancel_outlined),
+                  onPressed: () {
+                    bytes = null;
+                    Navigator.pop(context);
+                  },
+                ),
+                IconButton(
+                  icon: const Icon(Icons.done),
+                  onPressed: () {
+                    _controller.crop();
+                  },
+                )
+              ],
+            ),
+          ),
+        );
+      },
     ),
   );
+
+  if (bytes == null) {
+    return null;
+  }
+
+  final Directory tempDir = await getTemporaryDirectory();
+  final String tempPath = tempDir.path;
+  final String filePath = '$tempPath/upload_img_file_01.tmp';
```

Comment:
Should probably be deleted after the upload, though I understand that it will overridden by the next image.
Btw it's always just one upload at a time?

Reply:
I just checked we cannot delete the file directly, otherwise it will fail uploading, becouse it can't finde the file. I think that with the file object we only pass some kind of pointer